types/fetch.d.ts

```diff
@@ -0,0 +1,132 @@
+  data: AsyncIterable<Uint8Array>
+  disturbed: boolean
+  readonly [Symbol.asyncIterator]: () => AsyncIterator<Uint8Array>
+}
+
+export type BodyInit =
+  | ArrayBuffer
+  | AsyncIterable<Uint8Array>
+  | Blob
+  | FormData
+  | Iterable<Uint8Array>
+  | NodeJS.ArrayBufferView
+  | URLSearchParams
+  | null
+  | string
+
+export interface BodyMixin {
+  readonly body: ControlledAsyncIterable | null
+  readonly bodyUsed: boolean
+
+  readonly arrayBuffer: () => Promise<Buffer>
+  readonly json: () => Promise<unknown>
+  readonly text: () => Promise<string>
+}
+
+export type HeadersInit = Iterable<[string, string]> | Record<string, string>
+
+export declare class Headers implements Iterable<[string, string]> {
+  constructor (init?: HeadersInit)
+  readonly append: (name: string, value: string) => void
+  readonly delete: (name: string) => void
+  readonly get: (name: string) => string | null
+  readonly has: (name: string) => boolean
+  readonly set: (name: string, value: string) => void
+  readonly forEach: (
+    callbackfn: (value: string, key: string, iterable: Headers) => void,
+    thisArg?: unknown
+  ) => void
+
+  readonly keys: () => IterableIterator<string>
+  readonly values: () => IterableIterator<string>
+  readonly entries: () => IterableIterator<[string, string]>
+  readonly [Symbol.iterator]: () => Iterator<[string, string]>
+}
+
+export interface RequestInit {
+  readonly method?: string
+  readonly keepalive?: boolean
+  readonly headers?: HeadersInit
+  readonly body?: BodyInit
+  readonly redirect?: RequestRedirect
+  readonly integrity?: string
+  readonly signal?: AbortSignal
+}
+
+export type RequestRedirect = 'error' | 'follow' | 'manual'
+
+export declare class Request implements BodyMixin {
+  constructor (input: RequestInfo, init?: RequestInit)
+
+  readonly url: string
+  readonly method: string
+  readonly headers: Headers
+  readonly redirect: string
+  readonly integrity: string
+  readonly keepalive: boolean
+  readonly signal: AbortSignal
+
+  readonly body: ControlledAsyncIterable | null
+  readonly bodyUsed: boolean
+
+  readonly arrayBuffer: () => Promise<Buffer>
+  readonly json: () => Promise<unknown>
+  readonly text: () => Promise<string>
+
+  readonly clone: () => Request
+}
+
+export interface ResponseInit {
+  readonly status?: number
+  readonly statusText?: string
+  readonly headers?: HeadersInit
+}
+
+export type ResponseType =
+	| 'basic'
+	| 'cors'
+	| 'default'
+	| 'error'
+	| 'opaque'
+	| 'opaqueredirect'
+
+export declare class Response implements BodyMixin {
+  constructor (body?: BodyInit, init?: ResponseInit)
+
+  readonly headers: Headers
+  readonly ok: boolean
+  readonly status: number
+  readonly statusText: string
+  readonly type: ResponseType
+  readonly url: string
+  readonly redirected: boolean
+
+  readonly body: ControlledAsyncIterable | null
+  readonly bodyUsed: boolean
+
+  readonly arrayBuffer: () => Promise<Buffer>
+  readonly json: () => Promise<unknown>
```

Comment:
I just want to point out that this will break type compatibility with other packages & the native types.

```ts
const r = await fetch('https://some-json-api.com/');
const b = await r.json();

// in node-fetch, native types, etc (without casts)
b.something.somethingelse.anotherthing
// with this
b.something
  ^ 'cannot read property "something" of unknown'

```

`unknown` is a much better type than `any`, it adds an extra step for those transitioning from another package. Not sure if it really matters to you guys but I just wanted to point it out. 👍

Reply:
`node-fetch@3.0.0-beta.10` uses `Promise<unknown>` as well, and switching to `Promise<any>` wouldn't be breaking